pkg/proto/v0/settings.pb.micro_test.go

```diff
@@ -0,0 +1,272 @@
+			"",
+			"simple-display-name",
+			"simple-extension-name",
+			CustomError{
+				Id:     "go.micro.client",
+				Code:   500,
+				Detail: "key cannot be empty",
+				Status: "Internal Server Error",
+			},
+		},
+		{
+			"display name missing",
+			"simple-key",
+			"",
+			"simple-extension-name",
+			CustomError{},
+		},
+		{
+			"extension name missing",
+			"simple-key",
+			"simple-display-name",
+			"",
+			CustomError{},
+		},
+	}
+	for _, testCase := range tests {
+		t.Run(testCase.testDataName, func(t *testing.T) {
+			bundle := proto.SettingsBundle{
+				Key:         testCase.Key,
+				DisplayName: testCase.DisplayName,
+				Extension:   testCase.Extension,
+				Settings:    nil,
+			}
+			createRequest := proto.CreateSettingsBundleRequest{
+				SettingsBundle: &bundle,
+			}
+
+			client := service.Client()
+			cl := proto.NewBundleService("com.owncloud.api.settings", client)
+
+			cresponse, err := cl.CreateSettingsBundle(context.Background(), &createRequest)
+			if err != nil || (CustomError{} != testCase.expectedError) {
+				var errorData CustomError
+				json.Unmarshal([]byte(err.Error()), &errorData)
+				assert.Equal(t, testCase.expectedError.Id, errorData.Id)
+				assert.Equal(t, testCase.expectedError.Code, errorData.Code)
+				assert.Equal(t, testCase.expectedError.Detail, errorData.Detail)
+				assert.Equal(t, testCase.expectedError.Status, errorData.Status)
+			} else {
+				assert.Equal(t, testCase.Key, cresponse.SettingsBundle.Key)
+				assert.Equal(t, testCase.DisplayName, cresponse.SettingsBundle.DisplayName)
+				assert.Equal(t, testCase.Extension, cresponse.SettingsBundle.Extension)
+			}
+		})
+	}
+}
+
+func TestListSettingsBundlesWithNoSettings(t *testing.T) {
+	var tests = []TestStruct{
+		{
+			"ASCII",
+			"simple-key",
+			"simple-display-name",
+			"simple-extension-name",
+			CustomError{},
+		},
+		{
+			"space in values",
+			"simple key",
+			"simple display name",
+			"simple extension name",
+			CustomError{},
+		},
+		{
+			"UNICODE",
+			"सिम्पलेछबि",
+			"सिम्पले-name",
+			"सिम्पले-extension-name",
+			CustomError{},
+		},
+	}
+	for _, testCase := range tests {
+		t.Run(testCase.testDataName, func(t *testing.T) {
+			bundle := proto.SettingsBundle{
+				Key:         testCase.Key,
+				DisplayName: testCase.DisplayName,
+				Extension:   testCase.Extension,
+				Settings:    nil,
+			}
+			createRequest := proto.CreateSettingsBundleRequest{
+				SettingsBundle: &bundle,
+			}
+
+			client := service.Client()
+			cl := proto.NewBundleService("com.owncloud.api.settings", client)
+
+			_, err := cl.CreateSettingsBundle(context.Background(), &createRequest)
+			assert.NoError(t, err)
+			request := proto.ListSettingsBundlesRequest{Extension: testCase.Extension}
+			response, err := cl.ListSettingsBundles(context.Background(), &request)
+			assert.NoError(t, err)
+			assert.Equal(t, testCase.Key, response.SettingsBundles[0].Key)
+			assert.Equal(t, "", response.SettingsBundles[0].DisplayName)
```

Comment:
here the DisplayName is always empty. Is that correct?

Reply:
Heavily in progress. For now the actual settings bundles are not parsed from the files, only from the filenames. Filenames only contain key and extension, so because of that only those two values are returned. We have to think about performance when reading all existing settings bundles for a `list` call...